.github/workflows/github-action-lint.yaml

```diff
@@ -0,0 +1,37 @@
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_call:
+
+name: Lint (GitHub Actions)
+
+jobs:
+  check-github-actions:
+    name: Check GitHub Actions
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check for affected paths
+        id: changes
+        uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        with:
+          filters: |
+            github-actions:
+              - '.github/*/**/*.{yaml,yml}'
+
+      - name: Ensure SHA pinned actions
+        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@b35f285b9bb7e80de0967367cee66d3b6d50ceca # v3.0.1
```

Comment:
Same as the above.

Reply:
https://github.com/gravitational/github-terraform/pull/733